### DISCUSSION TITLE
Channel-specific management permissions

### DISCUSSION THREAD

**Discussion by akhial** [saleor/saleor#10883] [Ideas]
Hi Saleor team!

Saleor currently doesn't support restricting management permissions on a per-channel basis.
Consider a scenario where a store has a high number of warehouses with a high number of products per warehouse. This would require multiple warehouse managers that keep tabs on stock levels and incoming SKUs. Possibly one manager per warehouse.

Currently Saleor can only restrict permissions for all channels at once, so all warehouse managers can only have management access to all warehouses and channels or none at all.
This leads to a lot of confusion and poor security. And hinders adoption of Saleor at scale.

Maybe this can be achieved by adding a `channel` configuration to permission groups?
https://docs.saleor.io/docs/3.x/developer/api-reference/objects/group

Thanks for your work,

**Comment by peteror** [comment]
Hi, I believe this is already under construction / consideration https://github.com/saleor/saleor/pull/10423

**Reply by timuric** [reply]
Yes, we are actively working on it
